component/newstorage/edv/models.go

```diff
@@ -0,0 +1,96 @@
+type encryptedDocument struct {
+	ID                          string                       `json:"id"`
+	Sequence                    int                          `json:"sequence"`
+	IndexedAttributeCollections []indexedAttributeCollection `json:"indexed,omitempty"`
+	JWE                         json.RawMessage              `json:"jwe"`
+}
+
+// indexedAttributeCollection represents a collection of indexed attributes,
+// all of which share a common MAC algorithm and key.
+// This format is based on https://identity.foundation/confidential-storage/#creating-encrypted-indexes.
+type indexedAttributeCollection struct {
+	Sequence          int                `json:"sequence"`
+	HMAC              idTypePair         `json:"hmac"`
+	IndexedAttributes []indexedAttribute `json:"attributes"`
+}
+
+// indexedAttribute represents a single indexed attribute.
+type indexedAttribute struct {
+	Name   string `json:"name"`
+	Value  string `json:"value"`
+	Unique bool   `json:"unique"`
+}
+
+// idTypePair represents an ID+Type pair.
+// TODO: #2262 This is a simplified version of the actual EDV query format, which is still not finalized
+//  in the spec as of writing. See: https://github.com/decentralized-identity/confidential-storage/issues/34.
+type idTypePair struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
+
+// nameAndValueQuery represents a name+value pair that can be used to query the encrypted indices for specific data.
+// TODO: #2262 This is a simplified version of the actual EDV query format, which is still not finalized
+//  in the spec as of writing. See: https://github.com/decentralized-identity/confidential-storage/issues/34.
+// ReturnFullDocuments is currently non-standard and should only be used with an EDV server that supports it.
+type nameAndValueQuery struct {
+	ReturnFullDocuments bool   `json:"returnFullDocuments"`
+	Name                string `json:"index"`
+	Value               string `json:"equals"`
+}
+
+// hasQuery represents a simpler version of query above that matches all documents that are tagged with the index name
+// specified in "has", regardless of index value.
+// ReturnFullDocuments is currently non-standard and should only be used with an EDV server that supports it.
+type hasQuery struct {
+	ReturnFullDocuments bool   `json:"returnFullDocuments"`
+	Has                 string `json:"has"`
+}
+
+// batch represents a batch of operations to be performed in a vault.
+type batch []vaultOperation
+
+const (
+	// upsertDocumentVaultOperation represents an upsert operation to be performed in a batch.
+	upsertDocumentVaultOperation = "upsert"
+	// deleteDocumentVaultOperation represents a delete operation to be performed in a batch.
+	deleteDocumentVaultOperation = "delete"
+)
+
+// vaultOperation represents an upsert or delete operation to be performed in a vault.
+// This is currently non-standard and should only be used with an EDV server that supports it.
+type vaultOperation struct {
```

Comment:
The format of this vault operation will probably change when (if?) a batch endpoint gets added to the EDV spec.